xray/plot/plot.py

```diff
@@ -386,8 +406,12 @@ def _plot2d(plotfunc):
         The mapping from data values to color space. If not provided, this
         will be either be ``viridis`` (if the function infers a sequential
         dataset) or ``RdBu_r`` (if the function infers a diverging dataset).
-        When ``levels`` is provided and when `Seaborn` is installed, ``cmap``
-        may also be a `seaborn` color palette or a list of colors.
+        When when `Seaborn` is installed, ``cmap`` may also be a `seaborn` 
+        color palette. If ``cmap`` is seaborn color palette and the plot type
+        is not ``contour`` or ``contourf``, ``levels`` must also be specified.
```

Comment:
Seaborn colormaps (e.g. `husl`) need to have the number of levels specified. This is the only way in which they behave differently from matplotlib colormaps. This is a little bit annoying, since you can do ``imshow(cmap='jet')`` but instead have to do ``imshow(cmap='husl', levels=20)``. I have added a test for this.